src/result_processor.c

```diff
@@ -44,8 +44,10 @@ static int UnlockSpec_and_ReturnRPResult(RedisSearchCtx *sctx, int result_status
 typedef struct {
   ResultProcessor base;
   QueryIterator *iterator;
-  size_t timeoutLimiter;    // counter to limit number of calls to TimedOut_WithCounter()
   RedisSearchCtx *sctx;
+  uint32_t timeoutLimiter;    // counter to limit number of calls to TimedOut_WithCounter()
+  uint32_t slotsVersion;      // version of the slot ranges used for filtering
+  const RedisModuleSlotRangeArray *slotRanges_; // Query slot ranges info, may be used for filtering
   const SharedSlotRangeArray *slotRanges; // Owned slot ranges info, may be used for filtering
```

Comment:
These pairs of `slotRanges` and `slotRanges_` are somewhat confusing; let's try to find a better naming.
Also, do we need to hold both slot range lists at the `RPQueryIterator`? Aren't we always filtering by the query slot range info that came from the coord (as long as the local slots are a subset of those)?